internal/api/serve_test.go

```diff
@@ -0,0 +1,305 @@
+
+		assert.EventuallyWithT(t, func(c *assert.CollectT) {
+			assert.Len(c, cr.Counters(t).Kvs, 1)
+		}, time.Second*3, time.Millisecond*10)
+	})
+
+	t.Run("deleting with empty string should delete all jobs", func(t *testing.T) {
+		t.Parallel()
+
+		cr := cron.SinglePartitionRun(t)
+
+		require.NoError(t, cr.Add(context.Background(), "a1", &api.Job{Schedule: ptr.Of("@every 1s")}))
+		require.NoError(t, cr.Add(context.Background(), "b2", &api.Job{Schedule: ptr.Of("@every 1s")}))
+		require.NoError(t, cr.Add(context.Background(), "c3", &api.Job{Schedule: ptr.Of("@every 1s")}))
+
+		assert.Len(t, cr.Jobs(t).Kvs, 3)
+		assert.EventuallyWithT(t, func(c *assert.CollectT) {
+			assert.Len(c, cr.Counters(t).Kvs, 3)
+		}, time.Second*3, time.Millisecond*10)
+
+		require.NoError(t, cr.DeletePrefixes(context.Background(), ""))
+
+		assert.Empty(t, cr.Jobs(t).Kvs)
+
+		assert.EventuallyWithT(t, func(c *assert.CollectT) {
+			assert.Empty(c, cr.Counters(t).Kvs)
+		}, time.Second*3, time.Millisecond*10)
+	})
+
+	t.Run("deleting all with prefix in cluster should delete all jobs with prefix", func(t *testing.T) {
+		t.Parallel()
+
+		cr := cron.TripplePartitionRun(t)
+
+		for i := 0; i < 100; i++ {
+			require.NoError(t, cr.Add(context.Background(), "a"+strconv.Itoa(i), &api.Job{Schedule: ptr.Of("@every 1s")}))
+		}
+
+		assert.Len(t, cr.Jobs(t).Kvs, 100)
+		assert.EventuallyWithT(t, func(c *assert.CollectT) {
+			assert.Len(c, cr.Counters(t).Kvs, 100)
+		}, time.Second*10, time.Millisecond*10)
+
+		require.NoError(t, cr.DeletePrefixes(context.Background(), "a"))
+
+		assert.Empty(t, cr.Jobs(t).Kvs)
+
+		assert.EventuallyWithT(t, func(c *assert.CollectT) {
+			assert.Empty(c, cr.Counters(t).Kvs)
+		}, time.Second*10, time.Millisecond*10)
+
+		calls := cr.Calls.Load()
+		time.Sleep(time.Second * 2)
+		assert.Equal(t, calls, cr.Calls.Load())
+	})
+
+	t.Run("deleting no prefixes should delete nothing", func(t *testing.T) {
+		t.Parallel()
+
+		cr := cron.SinglePartitionRun(t)
+
+		require.NoError(t, cr.Add(context.Background(), "a1", &api.Job{Schedule: ptr.Of("@every 1s")}))
+		require.NoError(t, cr.Add(context.Background(), "b2", &api.Job{Schedule: ptr.Of("@every 1s")}))
+		require.NoError(t, cr.Add(context.Background(), "c3", &api.Job{Schedule: ptr.Of("@every 1s")}))
+
+		assert.Len(t, cr.Jobs(t).Kvs, 3)
+		assert.EventuallyWithT(t, func(c *assert.CollectT) {
+			assert.Len(c, cr.Counters(t).Kvs, 3)
+		}, time.Second*3, time.Millisecond*10)
+
+		require.NoError(t, cr.DeletePrefixes(context.Background()))
+
+		assert.Len(t, cr.Jobs(t).Kvs, 3)
+		assert.Len(t, cr.Counters(t).Kvs, 3)
+	})
+}
+
+func Test_List(t *testing.T) {
+	t.Parallel()
+
+	t.Run("List with no jobs should return empty", func(t *testing.T) {
+		t.Parallel()
+
+		cron := cron.SinglePartitionRun(t)
+
+		resp, err := cron.List(context.Background(), "")
+		require.NoError(t, err)
+		assert.Empty(t, resp.Jobs)
+	})
+
+	t.Run("List should return jobs which are in the namespace", func(t *testing.T) {
```

Comment:
This test is good for now. Have you done any manual validation as to what happens when the list returns too large of a list of jobs? I imagine we want to do a quick sanity check on this for the auto migration of reminders.

Reply:
Other than this last comment, the PR lgtm

Reply:
Thanks for picking this up. I've been looking into the code and documentation and as far as I see, `more` is only relevant in the circumstances where a `limit` is set and the response list exceeds that. Since we set no limit (`limit=0`), this value will never be `true`. I'm going to go ahead and remove it from the protos. Thanks again!